libs/content/workspace/src/tab/markdown_editor/bounds.rs

```diff
@@ -266,31 +263,35 @@ pub fn calc_links(buffer: &SubBuffer, text: &Text, ast: &Ast) -> PlainTextLinks
 }
 
 pub fn captured(
-    cursor: Cursor, paragraphs: &Paragraphs, ast: &Ast, text_range: &AstTextRange,
-    hover_syntax_reveal_debounce_state: HoverSyntaxRevealDebounceState, appearance: &Appearance,
-    cursor_paragraphs: (usize, usize),
+    cursor: Cursor, paragraphs: &Paragraphs, ast: &Ast, ast_text_range: &AstTextRange,
+    hover_syntax_reveal_debounce_state: &HoverSyntaxRevealDebounceState, appearance: &Appearance,
 ) -> bool {
-    let ast_node_range = ast.nodes[*text_range.ancestors.last().unwrap()].range;
-    let intersects_selection = ast_node_range.intersects_allow_empty(&cursor.selection);
-
-    let debounce_satisfied = hover_syntax_reveal_debounce_state.pointer_offset_updated_at
-        < Instant::now() - HOVER_SYNTAX_REVEAL_DEBOUNCE;
-    let intersects_pointer = debounce_satisfied
-        && hover_syntax_reveal_debounce_state
-            .pointer_offset
-            .map(|pointer_offset| {
-                ast_node_range.intersects(&(pointer_offset, pointer_offset), true)
-            })
-            .unwrap_or(false);
-
-    let text_range_paragraphs = paragraphs.find_intersecting(text_range.range, true);
-    let in_capture_disabled_paragraph = appearance.markdown_capture_disabled_for_cursor_paragraph
-        && cursor_paragraphs.intersects(&text_range_paragraphs, false);
-
-    match appearance.markdown_capture(text_range.node(ast).node_type()) {
+    if ast_text_range.range_type == AstTextRangeType::Text {
+        return false;
+    }
```

Comment:
Moved into this fn from caller; this was one option for reducing how often the rest of the logic in this function was invoked. I did it to help debugging but I suppose it improves performance as well in a way I didn't bother to measure